sky/shell/platform/ios/accessibility_bridge.h

```diff
@@ -0,0 +1,166 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SKY_SHELL_PLATFORM_IOS_SEMANTICS_LISTENER_IMPL_H_
+#define SKY_SHELL_PLATFORM_IOS_SEMANTICS_LISTENER_IMPL_H_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/macros.h"
+#include "base/memory/weak_ptr.h"
+#include "mojo/public/cpp/bindings/array.h"
+#include "mojo/public/cpp/bindings/strong_binding.h"
+#include "sky/engine/platform/geometry/FloatRect.h"
+#include "sky/services/semantics/semantics.mojom.h"
+#include "sky/shell/platform/ios/FlutterView.h"
+#include "third_party/skia/include/core/SkRect.h"
+#include "third_party/skia/include/utils/SkMatrix44.h"
+
+namespace sky {
+namespace shell {
+namespace a11y {
+
+class Node;
+
+typedef std::shared_ptr<Node> NodePtr;
+
+// Class that listens for updates to the semantic tree from Dart code and issues
+// corresponding updates to the FlutterView's accessibility elements.
+class AccessibilityBridge final : public semantics::SemanticsListener {
+ public:
+  AccessibilityBridge(FlutterView*, semantics::SemanticsServerPtr&);
+  ~AccessibilityBridge() override;
+
+  // Listener method invoked from Dart that signals that the specified semantic
+  // nodes have been updated in the render tree.
+  void UpdateSemanticsTree(mojo::Array<semantics::SemanticsNodePtr>) override;
+
+  base::WeakPtr<AccessibilityBridge> AsWeakPtr();
+
+ private:
+  typedef std::map<long, NodePtr> NodeMap;
+  friend class Node;
+
+  NodePtr UpdateNode(const semantics::SemanticsNodePtr& node);
+  void RemoveNode(NodePtr node);
+
+  NSArray* CreateAccessibleElements() const;
+
+  // We expect to have the same lifetime as the view
+  FlutterView* view_;
+  NodeMap nodes_;
+
+  mojo::StrongBinding<semantics::SemanticsListener> binding_;
+
+  base::WeakPtrFactory<AccessibilityBridge> weak_factory_;
+
+  DISALLOW_COPY_AND_ASSIGN(AccessibilityBridge);
+};
+
+struct Flags {
+  Flags()
+      : can_be_tapped(FALSE),
```

Comment:
Huh, looks like lower and upper case are both defined natively.  But lower looks to be our style, so changed it to that.